crates/nu-parser/src/parser.rs

```diff
@@ -1877,6 +1877,22 @@ pub fn parse_duration(
 ) -> (Expression, Option<ParseError>) {
     trace!("parsing: duration");
 
+    let bytes = working_set.get_span_contents(span);
+
+    match parse_duration_bytes(bytes, span) {
+        Some(expression) => (expression, None),
+        None => (
+            garbage(span),
+            Some(ParseError::Mismatch(
+                "duration".into(),
+                "non-duration unit".into(),
+                span,
+            )),
+        ),
+    }
+}
+
+pub fn parse_duration_bytes(bytes: &[u8], span: Span) -> Option<Expression> {
```

Comment:
I'm not sure about the return type. Is `Result<Expression, ParserError>` better?